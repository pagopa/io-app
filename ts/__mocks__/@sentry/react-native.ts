export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
