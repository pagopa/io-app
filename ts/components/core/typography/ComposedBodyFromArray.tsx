import React from "react";
import {
  Body,
  ExternalTypographyProps,
  IOColors,
  IOFontWeight,
  IOTheme,
  TypographyProps
} from "@pagopa/io-app-design-system";

type PartialAllowedColors = Extract<
  IOColors,
  "bluegreyDark" | "white" | "blue" | "bluegrey" | "bluegreyLight"
>;
type AllowedColors = PartialAllowedColors | IOTheme["textBody-default"];
type AllowedWeight = IOFontWeight | "Regular" | "Semibold";

export type BodyProps = ExternalTypographyProps<
  TypographyProps<AllowedWeight, AllowedColors> & {
    text: string | React.ReactElement;
  }
>;

type PropsComposedBody = {
  body: Array<BodyProps>;
  textAlign?: "auto" | "left" | "right" | "center" | "justify" | undefined;
};

export const ComposedBodyFromArray = ({
  body,
  textAlign = "center"
}: PropsComposedBody) => (
  <Body style={{ textAlign }}>
    {body.map(({ text, key, ...props }, i) => (
      <Body key={key ?? `OpResScreen_${i}`} {...props}>
        {text}
      </Body>
    ))}
  </Body>
);
