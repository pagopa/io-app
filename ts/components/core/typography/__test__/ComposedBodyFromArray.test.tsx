import React from "react";
import { render } from "@testing-library/react-native";
import { BodyProps, ComposedBodyFromArray } from "../ComposedBodyFromArray";

describe("ComposedBodyFromArray", () => {
  it("should match snapshot, empty body, default text align", () => {
    const component = render(<ComposedBodyFromArray body={[]} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, empty body, auto text align", () => {
    const component = render(
      <ComposedBodyFromArray body={[]} textAlign="auto" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, empty body, center text align", () => {
    const component = render(
      <ComposedBodyFromArray body={[]} textAlign="center" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, empty body, justify text align", () => {
    const component = render(
      <ComposedBodyFromArray body={[]} textAlign="justify" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, empty body, left text align", () => {
    const component = render(
      <ComposedBodyFromArray body={[]} textAlign="left" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, empty body, right text align", () => {
    const component = render(
      <ComposedBodyFromArray body={[]} textAlign="right" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, multiple items body, default text align", () => {
    const component = render(
      <ComposedBodyFromArray body={multipleItemsBody} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, multiple items body, auto text align", () => {
    const component = render(
      <ComposedBodyFromArray body={multipleItemsBody} textAlign="auto" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, multiple items body, center text align", () => {
    const component = render(
      <ComposedBodyFromArray body={multipleItemsBody} textAlign="center" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, multiple items body, justify text align", () => {
    const component = render(
      <ComposedBodyFromArray body={multipleItemsBody} textAlign="justify" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, multiple items body, left text align", () => {
    const component = render(
      <ComposedBodyFromArray body={multipleItemsBody} textAlign="left" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should match snapshot, multiple items body, right text align", () => {
    const component = render(
      <ComposedBodyFromArray body={multipleItemsBody} textAlign="right" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  const multipleItemsBody = [
    {
      text: "Multiple ",
      weight: "Regular"
    },
    {
      text: "styles",
      weight: "Bold"
    },
    {
      text: " text ",
      weight: "Light"
    },
    {
      text: "just",
      weight: "Medium"
    },
    {
      text: " weights",
      weight: "SemiBold"
    }
  ] as Array<BodyProps>;
});
