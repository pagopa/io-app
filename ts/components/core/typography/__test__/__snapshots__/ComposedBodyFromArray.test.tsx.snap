// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposedBodyFromArray should match snapshot, empty body, auto text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "auto",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
/>
`;

exports[`ComposedBodyFromArray should match snapshot, empty body, center text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
/>
`;

exports[`ComposedBodyFromArray should match snapshot, empty body, default text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
/>
`;

exports[`ComposedBodyFromArray should match snapshot, empty body, justify text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "justify",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
/>
`;

exports[`ComposedBodyFromArray should match snapshot, empty body, left text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "left",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
/>
`;

exports[`ComposedBodyFromArray should match snapshot, empty body, right text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "right",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
/>
`;

exports[`ComposedBodyFromArray should match snapshot, multiple items body, auto text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "auto",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Multiple 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "700",
        },
      ]
    }
    weight="Bold"
  >
    styles
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "300",
        },
      ]
    }
    weight="Light"
  >
     text 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      ]
    }
    weight="Medium"
  >
    just
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
     weights
  </Text>
</Text>
`;

exports[`ComposedBodyFromArray should match snapshot, multiple items body, center text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Multiple 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "700",
        },
      ]
    }
    weight="Bold"
  >
    styles
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "300",
        },
      ]
    }
    weight="Light"
  >
     text 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      ]
    }
    weight="Medium"
  >
    just
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
     weights
  </Text>
</Text>
`;

exports[`ComposedBodyFromArray should match snapshot, multiple items body, default text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Multiple 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "700",
        },
      ]
    }
    weight="Bold"
  >
    styles
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "300",
        },
      ]
    }
    weight="Light"
  >
     text 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      ]
    }
    weight="Medium"
  >
    just
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
     weights
  </Text>
</Text>
`;

exports[`ComposedBodyFromArray should match snapshot, multiple items body, justify text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "justify",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Multiple 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "700",
        },
      ]
    }
    weight="Bold"
  >
    styles
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "300",
        },
      ]
    }
    weight="Light"
  >
     text 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      ]
    }
    weight="Medium"
  >
    just
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
     weights
  </Text>
</Text>
`;

exports[`ComposedBodyFromArray should match snapshot, multiple items body, left text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "left",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Multiple 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "700",
        },
      ]
    }
    weight="Bold"
  >
    styles
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "300",
        },
      ]
    }
    weight="Light"
  >
     text 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      ]
    }
    weight="Medium"
  >
    just
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
     weights
  </Text>
</Text>
`;

exports[`ComposedBodyFromArray should match snapshot, multiple items body, right text align 1`] = `
<Text
  allowFontScaling={false}
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="TitilliumWeb"
  fontStyle={
    Object {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  style={
    Array [
      Object {
        "textAlign": "right",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {
        "color": "#475A6D",
        "fontFamily": "Titillium Web",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Multiple 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "700",
        },
      ]
    }
    weight="Bold"
  >
    styles
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "300",
        },
      ]
    }
    weight="Light"
  >
     text 
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "500",
        },
      ]
    }
    weight="Medium"
  >
    just
  </Text>
  <Text
    allowFontScaling={false}
    color="bluegrey"
    defaultColor="bluegrey"
    defaultWeight="Regular"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
     weights
  </Text>
</Text>
`;
